daemon/list_test.go

```diff
@@ -79,7 +81,46 @@ func containerListContainsName(containers []*containertypes.Summary, name string
 	return false
 }
 
-func TestListInvalidFilter(t *testing.T) {
+func TestContainerList(t *testing.T) {
+	db, err := container.NewViewDB()
+	assert.NilError(t, err)
+	d := &Daemon{
+		containersReplica: db,
+	}
+
+	// test list with different number of containers
+	for _, num := range []int{0, 1, 2, 4, 8, 16, 32, 64, 100} {
+		t.Run(fmt.Sprintf("%d containers", num), func(t *testing.T) {
+			db, err := container.NewViewDB() // new DB to ignore prior containers
+			assert.NilError(t, err)
+			d = &Daemon{
+				containersReplica: db,
+			}
+
+			// create the containers
+			containers := make([]*container.Container, num)
+			for i := range num {
+				name := fmt.Sprintf("cont-%d", i)
+				containers[i] = setupContainerWithName(t, name, d)
+				// ensure container timestamps are separated enough so the
+				// sort used by d.Containers() can deterministically sort them.
+				containers[i].Created = containers[i].Created.Add(time.Millisecond)
```

Comment:
🤔 😅 So, erm, are we actually adding separation here, or are we just shifting all containers by 1 millisecond ? 🙈 

I guess this could be either;

add 1 millisecond to the previous container

```suggestion
				if i > 0 {
					containers[i].Created = containers[i-1].Created.Add(time.Millisecond)
				}
```

Or we could just add "i" milliseconds;

```suggestion
				containers[i].Created = containers[i].Created.Add(time.Duration(i) * time.Millisecond)
```

Reply:
Oh yes, thanks @thaJeztah!